src/Packet.cpp

```diff
@@ -0,0 +1,9 @@
+#include "Packet.hpp"
+
+Packet::Packet(std::uint64_t sender_id, std::uint64_t sending_time_ns, std::uint64_t delivery_time_ns)
+    : sender_id(sender_id), sending_time_ns(sending_time_ns), estimated_delivery_time_ns(delivery_time_ns) { }
+
+// The earlier the estimated_delivery_time, the faster the packet will arrive
+bool Packet::operator<(const Packet& other) const {
+    return estimated_delivery_time_ns > other.estimated_delivery_time_ns;
```

Comment:
This comparison is needed to sort packets in the priority queue, right? I wonder if this should be here... What about creating a separate class *Events* for this purpose?

Reply:
Yes, this comparison is needed to sort packets. I agree, class `Packet` contains several unsuitable variables like delivery time. It is a good idea to wrap `Packet` into `Event`.